openedxscorm/scormxblock.py

```diff
@@ -406,13 +406,21 @@ def extract_folder_base_path(self):
         Path to the folder where packages will be extracted.
         """
         return os.path.join(self.scorm_location(), self.location.block_id)
+    
+    def get_mode(self,data):
+        if('preview' in data['url']):
+            return "review"
+        return "normal"
 
     @XBlock.json_handler
     def scorm_get_value(self, data, _suffix):
         """
         Here we get only the get_value events that were not filtered by the LMSGetValue js function.
         """
         name = data.get("name")
+        if name in ["cmi.core.lesson_mode", "cmi.mode"]:
+            mode = self.get_mode(data)
```

Comment:
nit: can merge these two lines to simplify these
```suggestion
            return {"value": self.get_mode(data)}
```